src/odbc/unittests/connection_string_parse.c

```diff
@@ -0,0 +1,198 @@
+#include "common.h"
+#include <assert.h>
+#include "freetds/odbc.h"
+
+
+#ifdef _WIN32
+HINSTANCE hinstFreeTDS;
+#endif
+
+static void
+assert_equal_str(DSTR a, const char *b)
+{
+  assert(b && strncmp(tds_dstr_cstr(&a), b, tds_dstr_len(&a))==0);
```

Comment:
do you really want to test the beginning of the string or the entire one?

Reply:
I'm not sure what you mean by this, could you please elaborate?

Reply:
strncmp is comparing the beginning of the string. So `strncmp("foo", "foo123", 4)` is detecting the strings are the same